File: py-src/minicode/history.py
from __future__ import annotations

import json

from minicode.config import MINI_CODE_DIR, MINI_CODE_HISTORY_PATH


def load_history_entries() -> list[str]:
    if not MINI_CODE_HISTORY_PATH.exists():
        return []
    try:
        parsed = json.loads(MINI_CODE_HISTORY_PATH.read_text(encoding="utf-8"))
    except json.JSONDecodeError:
        return []
    entries = parsed.get("entries", [])
    return [str(entry) for entry in entries] if isinstance(entries, list) else []


def save_history_entries(entries: list[str]) -> None:
    MINI_CODE_DIR.mkdir(parents=True, exist_ok=True)
    MINI_CODE_HISTORY_PATH.write_text(
        json.dumps({"entries": entries[-200:]}, indent=2) + "\n",
        encoding="utf-8",
    )

File: ts-src/src/history.ts
import { mkdir, readFile, writeFile } from 'node:fs/promises'
import { MINI_CODE_DIR, MINI_CODE_HISTORY_PATH } from './config.js'

type HistoryFile = {
  entries: string[]
}

export async function loadHistoryEntries(): Promise<string[]> {
  try {
    const raw = await readFile(MINI_CODE_HISTORY_PATH, 'utf8')
    const parsed = JSON.parse(raw) as HistoryFile
    return Array.isArray(parsed.entries) ? parsed.entries : []
  } catch {
    return []
  }
}

export async function saveHistoryEntries(entries: string[]): Promise<void> {
  await mkdir(MINI_CODE_DIR, { recursive: true })
  await writeFile(
    MINI_CODE_HISTORY_PATH,
    `${JSON.stringify({ entries: entries.slice(-200) }, null, 2)}\n`,
    'utf8',
  )
}
